### PR TITLE
[MM-30269] Add MySQL collation note

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR adds a note to warn users that they should update their tables collation if they update to MySQL 8

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-30269

### CODE REVIEW

**Review by isacikgoz — APPROVED** [review]
Nice!

**Review by ethervoid — APPROVED** [review]
LGTM!

**Comment by isacikgoz** [comment]
For reference, this update will address the problem in https://github.com/mattermost/mattermost-server/issues/16206

File: source/install/requirements.rst [hunk 1]
```diff
@@ -113,11 +113,21 @@ Search limitations on MySQL:
...
   .. code-block:: text
-   
+
    [mysqld]
    default-authentication-plugin=mysql_native_password
-   
-In MySQL versions 8.0.0-8.0.11 ```ADMIN``` is a `reserved keyword <https://dev.mysql.com/doc/refman/8.0/en/keywords.html>`_, which is why our requirement for MySQL is version 8.0.12. 
+
+
+In MySQL 8, the default collation changed to ``utf8mb4_0900_ai_ci`` (https://dev.mysql.com/doc/mysqld-version-reference/en/optvar-changes-8-0.html). Therefore, if you update your MySQL installation to version 8, you will need to convert your database tables to use the new default collation:
```
Comment (justinegeffen) [inline source/install/requirements.rst:121]:
```suggestion
In MySQL 8, the default collation changed to ``utf8mb4_0900_ai_ci`` (https://dev.mysql.com/doc/mysqld-version-reference/en/optvar-changes-8-0.html). Therefore, if you update your MySQL installation to version 8, you'll need to convert your database tables to use the new default collation:
```

File: source/install/requirements.rst [hunk 2]
```diff
@@ -113,11 +113,21 @@ Search limitations on MySQL:
...
+
+.. code-block:: sql
+
+   ALTER TABLE table_name CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_0900_ai_ci;
+
```
Comment (justinegeffen) [inline source/install/requirements.rst:126]:
```suggestion
```

File: source/install/requirements.rst [hunk 3]
```diff
@@ -113,11 +113,21 @@ Search limitations on MySQL:
...
+
+Not making this change can cause the database to have tables with different collations, which will cause errors when executing queries.
```
Comment (justinegeffen) [inline source/install/requirements.rst:127]:
```suggestion
If this change isn't made, tables in the database may have different collations which will cause errors when executing queries.
```

File: source/install/requirements.rst [hunk 4]
```diff
@@ -113,11 +113,21 @@ Search limitations on MySQL:
...
+
+In MySQL versions 8.0.0-8.0.11 ```ADMIN``` is a `reserved keyword <https://dev.mysql.com/doc/refman/8.0/en/keywords.html>`_, which is why our requirement for MySQL is version 8.0.12.
```
Comment (justinegeffen) [inline source/install/requirements.rst:129]:
```suggestion
In MySQL versions 8.0.0 - 8.0.11 ``ADMIN`` is a `reserved keyword <https://dev.mysql.com/doc/refman/8.0/en/keywords.html>`_, which is why our requirement for MySQL is version 8.0.12.
```

File: source/install/requirements.rst (hunk 3 above)
Comment (mgdelacroix) [reply source/install/requirements.rst:127]: Could this be something like this?

```suggestion
If this change isn't made, tables in the database may end up having different collations which will cause errors when executing queries.
```

Intent here is to express that the problem doesn't happen immediately. If this change is not made, down the line when the user updates Mattermost to a version that creates tables, that's when the collations get mixed.

**Review by justinegeffen — APPROVED** [review]

**Comment by justinegeffen** [comment]
Thanks, @mgdelacroix, and thanks for reworking my suggestion. LGTM. :)

**Review by wiersgallak — APPROVED** [review]

**Pull request merged**